martin/src/file_config.rs

```diff
@@ -168,8 +168,16 @@ impl<T: ConfigExtras> FileConfigEnum<T> {
 
     pub fn finalize(&self, prefix: &str) -> UnrecognizedValues {
         let mut res = UnrecognizedValues::new();
+
         if let Self::Config(cfg) = self {
-            copy_unrecognized_config(&mut res, prefix, cfg.get_unrecognized());
+            let unrecognized = cfg.get_unrecognized();
+            copy_unrecognized_config(&mut res, prefix, unrecognized);
+
+            for (key, value) in unrecognized.into_iter() {
+                warn!(
+                    "Ignoring unrecognized configuration entry '{prefix}.{key}: {value}'. Please check your configuration file for typos."
+                );
```

Comment:
There is a clippy warning about `value` not having `Display` implemented.

Also, on second thought, we are not really objecting to `value`, rather we are objecting on the `key` not being something we recognise.

```suggestion
            for key in unrecognized.keys() {
                warn!(
                    "Ignoring unrecognized configuration key '{prefix}.{key}'. Please check your configuration file for typos."
                );
```